libs/category/src/effects/category.effects.ts

```diff
@@ -121,18 +121,18 @@ export class DaffCategoryEffects {
   toggleCategoryFilter$ : Observable<any> = this.actions$.pipe(
     ofType(DaffCategoryActionTypes.ToggleCategoryFilterAction),
     withLatestFrom(
-      this.store.pipe(select(selectSelectedCategoryId)),
-      this.store.pipe(select(selectCategoryPageSize)),
-      this.store.pipe(select(selectCategoryPageFilterRequests)),
-      this.store.pipe(select(selectCategoryPageAppliedSortOption)),
-      this.store.pipe(select(selectCategoryPageAppliedSortDirection))
+      this.store.pipe(select(getDaffCategorySelectors<U, V>().selectSelectedCategoryId)),
+      this.store.pipe(select(getDaffCategorySelectors<U, V>().selectCategoryPageSize)),
+      this.store.pipe(select(getDaffCategorySelectors<U, V>().selectCategoryPageFilterRequests)),
+      this.store.pipe(select(getDaffCategorySelectors<U, V>().selectCategoryPageAppliedSortOption)),
+      this.store.pipe(select(getDaffCategorySelectors<U, V>().selectCategoryPageAppliedSortDirection))
     ),
     switchMap((
 			[action, categoryId, pageSize, filterRequests, appliedSortOption, appliedSortDirection]: 
 			[DaffToggleCategoryFilter, string, number, DaffCategoryFilterRequest[], string, DaffSortDirectionEnum]
 		) => {
 			this.validateFilters(filterRequests);
-			return this.processCategoryGetRequest({
+			return this.processCategoryGetRequest(<T>{
```

Comment:
That's a really good question. I think the fact that it is needed means that these effects can never be truly generic, because of covariance/contravariance reasons.

Reply:
I need to think about it

Reply:
So this is definitely a problem, but I think I'll need to address it in a future PR, because the solution will be pretty extensive. Basically it needs to be casted, because it isn't correct. It's definitely a covariance/contravariance problem. This effect can only be guaranteed to work with a plain `DaffCategoryRequest`, but it's trying to work with any `T extends DaffCategoryRequest`. So if a user defines a new model with a required field outside of `DaffCategoryRequest`, then this will no longer work, because it will call their custom injected driver without the new required field. The only effect in this file that truly works is the first one that passes the whole request object. I think this PR should pass through for now though

Reply:
Yeah I think its the nature of these effects that they won't be generic. You pull data from selectors and build an object literal from it. I don't think that can be generic in any case but I also don't think that it *needs* to be generic. In any case, I think this is fine for now.

Reply:
@damienwebdev and I talked about this and we have a way forward (involves spreading the entire `T extends DaffCategoryRequest` over all of these effects), but it'll definitely be in another PR.